static/js/settings.js

```diff
@@ -55,14 +55,70 @@ const SettingsScreen = {
     this.availableAddons = new Map();
     this.installedAddons = new Map();
     this.fetchAddonDeferred = null;
+
+    this.insertTitleElement(this.menu, 'Settings',
+                            '/optimized-images/settings-icon.png');
+    this.insertTitleElement(this.domainSettings, 'Domain',
+                            '/optimized-images/domain-icon.png');
+    this.insertTitleElement(this.userSettingsMain, 'Users',
+                            '/optimized-images/users-icon.png');
+    this.insertTitleElement(this.userSettingsEdit, 'Edit User',
+                            '/optimized-images/user.svg');
+    this.insertTitleElement(this.userSettingsAdd, 'Add User',
+                            '/optimized-images/user.svg');
+    this.insertTitleElement(this.adapterSettings, 'Adapters',
+                            '/optimized-image/adapters-icon.png');
+    this.insertTitleElement(this.addonMainSettings, 'Add-ons',
+                            '/optimized-images/add-on.svg');
+    const addonConfigTitle =
+      this.insertTitleElement(this.addonConfigSettings,
+                              'Configure Add-on',
+                              '/optimized-images/add-on.svg');
+    this.addonConfigTitleName =
+      addonConfigTitle.querySelector('.section-title-name');
+
+    this.insertTitleElement(this.addonDiscoverySettings,
+                            'Discover New Add-ons',
+                            '/optimized-images/add-on.svg');
+    this.insertTitleElement(this.experimentSettings, 'Experiments',
+                            '/optimized-images/experiments-icon.png');
+    this.insertTitleElement(this.updateSettings, 'Updates',
+                            '/optimized-images/update-icon.svg');
+    this.insertTitleElement(this.authorizationSettings, 'Authorizations',
+                            '/optimized-images/authorization.svg');
+    this.insertTitleElement(this.developerSettings, 'Developer',
+                            '/optimized-images/developer-icon.svg');
+
```

Comment:
Is there are particular reason these headings need to be added dynamically rather than being part of the original HTML?

Reply:
Adding them to the original HTML would have been at least an hour worth of copy-pasting markup that we may want to change in the future (going to go into more detail about that below).